pages/staking/SelfLiquidation.tsx

```diff
@@ -0,0 +1,116 @@
+import useSynthetixQueries from '@synthetixio/queries';
+import Wei, { wei } from '@synthetixio/wei';
+import Button from 'components/Button';
+import { EXTERNAL_LINKS } from 'constants/links';
+import React, { useState } from 'react';
+import { useRecoilValue } from 'recoil';
+import TxConfirmationModal from 'sections/shared/modals/TxConfirmationModal';
+import { delegateWalletState, walletAddressState } from 'store/wallet';
+import styled from 'styled-components';
+import {
+	ExternalLink,
+	LineSpacer,
+	ModalContent,
+	ModalItem,
+	ModalItemTitle,
+	ModalItemText,
+} from 'styles/common';
+import { formatShortDateWithTime } from 'utils/formatters/date';
+import { formatPercent } from 'utils/formatters/number';
+import { Svg } from 'react-optimized-image';
+import WarningIcon from 'assets/svg/app/warning.svg';
+import { useTranslation } from 'react-i18next';
+import { Trans } from 'react-i18next';
+
+const SelfLiquidation: React.FC<{ percentageTargetCRatio: Wei }> = ({ percentageTargetCRatio }) => {
+	const { t } = useTranslation();
+	const { useGetLiquidationDataQuery, useSynthetixTxn } = useSynthetixQueries();
+	const [txModalOpen, setTxModalOpen] = useState<boolean>(false);
+
+	const walletAddress = useRecoilValue(walletAddressState);
+	const delegateWallet = useRecoilValue(delegateWalletState);
+	const addressToUse = delegateWallet?.address || walletAddress!;
+
+	const liquidationData = useGetLiquidationDataQuery(addressToUse);
+	const liquidationDeadlineForAccount =
+		liquidationData?.data?.liquidationDeadlineForAccount ?? wei(0);
+
+	const txn = useSynthetixTxn('Synthetix', 'liquidateSelf');
+	// You cant self liquidate with delegation
+	if (delegateWallet?.address) return null;
+	// You have not been flagged
+	if (liquidationDeadlineForAccount.eq(0)) return null;
+	return (
+		<>
+			<Container>
+				<Svg src={WarningIcon} />
+				<Title>{t('staking.flag-warning.title')}</Title>
+				<InfoText>
+					<Trans
+						i18nKey={'staking.flag-warning.info'}
+						components={[
+							<Strong />,
+							<StyledExternalLink href={EXTERNAL_LINKS.Synthetix.SIP148Liquidations} />,
+						]}
+						values={{
+							deadlineDate: formatShortDateWithTime(
+								liquidationDeadlineForAccount.toNumber() * 1000
+							),
+							percentageTargetCRatio: formatPercent(percentageTargetCRatio),
+						}}
+					/>
+				</InfoText>
+				<Button
+					style={{ marginBottom: 30 }}
```

Comment:
dunno what our policy is about inline styles?

Reply:
but if good for you, I'm good with :)

Reply:
Opsie, I'll change it